### PR TITLE
Enable virtual assistant for all applications

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary

Fixes https://github.com/finos/legend-studio/issues/1712
Depends on https://github.com/finos/legend/pull/566
## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

Legend Query:

https://user-images.githubusercontent.com/73408381/204555601-7b136199-ce4c-4da6-82a8-3123945eb223.mov

Legend Taxonomy:

https://user-images.githubusercontent.com/73408381/204555965-5b13198e-4ede-4115-811f-2e15d34dc6e6.mov

### CODE REVIEW

File: .changeset/spicy-vans-add.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+'@finos/legend-application': patch
+'@finos/legend-application-query': patch
+'@finos/legend-application-studio': patch
+'@finos/legend-application-taxonomy': patch
+'@finos/legend-query-builder': patch
+---
+Enable virtual assistant for all applications ([#1712](https://github.com/finos/legend-studio/issues/1712)).
```
Comment (akphi) [inline .changeset/spicy-vans-add.md:8]: This changeset should only apply for `@finos/legend-application` and it should be a `minor` change

Comment (akphi) [inline .changeset/spicy-vans-add.md:8]: This changeset should only apply for `@finos/legend-application` and it should be a `minor` change. Let's split up the changeset properly. Thanks!

File: packages/legend-application-query-bootstrap/scripts/setup.js [hunk 2]
```diff
@@ -77,6 +77,10 @@ export const setup = (outputDir) => {
               url: 'https://legend.finos.org/resource/studio/documentation/shared.json',
               simple: true,
             },
+            {
```
Comment (akphi) [inline packages/legend-application-query-bootstrap/scripts/setup.js:80]: like said, ideally, let's merge https://github.com/finos/legend/pull/566 first then we can also add https://legend.finos.org/resource/studio/documentation/dsl-service.json or something like that here. We should add this to other setup.js files as well

File: packages/legend-application-query/src/components/QuerySetup.tsx [hunk 3]
```diff
@@ -276,77 +277,99 @@ export const QuerySetupLandingPage = withQuerySetupLandingPageStore(
...
       );
     const showAllActionGroup = (): void => setupStore.setShowAllGroups(true);
-
+    const toggleAssistant = (): void =>
+      applicationStore.assistantService.toggleAssistant();
     useEffect(() => {
       setupStore.initialize(showAdvancedActions, showAllGroups, tagToFocus);
     }, [setupStore, showAdvancedActions, showAllGroups, tagToFocus]);
 
     return (
```
Comment (akphi) [inline packages/legend-application-query/src/components/QuerySetup.tsx:286]: I have to warn you one small thing here is that there are children screens of query setup (i.e. when you click on one of the modes), so we probably missed the status bar there

File: packages/legend-query-builder/src/components/QueryBuilder.tsx [hunk 4]
```diff
@@ -150,6 +153,20 @@ const QueryBuilderStatusBar = observer(
           >
             <HackerIcon />
           </button>
+          <button
```
Comment (akphi) [inline packages/legend-query-builder/src/components/QueryBuilder.tsx:156]: This is fine, but I also have to warn you of the potential consequence of this, in the embedded query builder, we now have the button to hide virtual assistant, that would make this button showing up twice in the UI (including the one in Legend Studio status bar)

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-application-query/src/components/QuerySetup.tsx (hunk 3 above)
Comment (YannanGao-gs) [reply packages/legend-application-query/src/components/QuerySetup.tsx:286]: status bar was added in the BasicQuerySetup, which will make sure there will be one status bar in every single children screen. (also double-checked by rendering each child component one by one) :) 
Screenshot 2022-11-29 at 11 57 55 AM

**Review by akphi — APPROVED** [review]

**Pull request merged**